libcodechecker/server/server.py

```diff
@@ -516,6 +518,9 @@ def connect(self):
             self.__connected = True
             self.__last_connect_attempt = None
             LOG.debug("Database connected.")
+
+            if self.__db_cleanup:
+                db_cleanup.run_cleanup_jobs(self.__session)
```

Comment:
Are we sure about this? This will mean that as long as the server is running, every "Product connected" action will result in the cleanup job running.

Reply:
Sorry, I don't understand this comment.

Reply:
At further reading of the code, I figured my fears are without base. You only set the value of `__db_cleanup` at server start.

I feared that every `CodeChecker cmd product add` call (which instantiates a `Product` object with this constructor here in question) will run a database clean, which would be superfluous operation.

Still, there is a hole. If I start the server with `--db-cleanup` and then add a product later on, its database is **not** cleaned up. (As this value is only propagated from the server's constructor method.)

Should this really be an option that starts a server and then keeps it running? Wouldn't shutting the server down after this operation be a better thing?

Reply:
The plan so far was to do cleanup only at server start. I don't know if the requirements will change about this, but we should take into consideration that other cleanup jobs will also take part here. That's why `run_cleanup_jobs()` function is created. Another action will be the one which resets severity levels in case the corresponding config file modifies (see #1075). This is a single action at server start.